### PR TITLE
feat(build): add windows and osx builds to Travis

### LINKED ISSUE #223: Add Travis CI Windows builds
See [blog announcement](https://blog.travis-ci.com/2018-10-11-windows-early-release).  It's an "Early Release" but worth testing.

### PR DESCRIPTION
**Pull request by moorereason**
Fixes #223 
Fixes #224

### CODE REVIEW

**Comment by moorereason** [comment]
It looks like most of the "advanced" stuff in the Makefile may be a problem for Travis windows builds.  I stripped a lot of that out to get it to build.

**Comment by xcoulon** [comment]
> It looks like most of the "advanced" stuff in the Makefile may be a problem for Travis windows builds. I stripped a lot of that out to get it to build.

Ok, I understand. Well, this makes the whole "build" much simpler, in the end, so it's even better :)
At this point, I believe we should add support for macos (`osx`), too. 

Please remove the `WIP` prefix on the PR title once you feel it's good to merge.

File: .travis.yml [hunk 1]
```diff
@@ -1,25 +1,28 @@
...
-  # - 1.7.x (abandonned - not supported by dep)
-  - "1.8.x"
-  - "1.9.x"
-  - "1.10.x"
+  #- "1.8.x"
+  #- "1.9.x"
+  #- "1.10.x"
   - "1.11.x"
-  - tip
+  #- tip
```
Comment (xcoulon) [inline .travis.yml]: did you have to remove all the older versions of `go` on purpose? are there limitations with regards to golang versions on Windows builders on Travis-ci ?

Comment (moorereason) [reply .travis.yml]: I removed the old versions to keep from hammering Travis while I tested.  I'll put them back before I'm done here.

**Comment by moorereason** [comment]
I rebased this PR to clean up the git log.  I'll update the OP to match the extended scope since we're adding OSX, too.

**Comment by xcoulon** [comment]
thanks for this PR, @moorereason ! The `Makefile` and the `.travis.yml` config file are now much simpler and having all platforms build on a single place is nice ;)

**Pull request merged**